python/test/unit/mesh/test_higher_order_mesh.py

```diff
@@ -36,10 +36,80 @@ def check_cell_volume(points, cell, domain, volume):
 
         mesh = create_mesh(MPI.COMM_WORLD, [ordered_cell], ordered_points, domain)
         area = assemble_scalar(form(1 * dx(mesh)))
-
         assert np.isclose(area, volume)
 
 
+@pytest.mark.skip_in_parallel
+@pytest.mark.parametrize('order', range(1, 5))
+def test_submesh(order):
+    points = []
+    points += [[i / order, j / order, 0] for j in range(order + 1)
+               for i in range(order + 1 - j)]
+    for k in range(1, order):
+        points += [[i / order, j / order + 0.1, k / order] for j in range(order + 1 - k)
+                   for i in range(order + 1 - k - j)]
+
+    points += [[0, 0, 1]]
+
+    def coord_to_vertex(x, y, z):
+        return z * (
+            3 * order ** 2 - 3 * order * z + 12 * order + z ** 2 - 6 * z + 11
+        ) // 6 + y * (2 * (order - z) + 3 - y) // 2 + x
+
+    # Define a cell using DOLFINx ordering
+    cell = [coord_to_vertex(x, y, z) for x, y, z in [
+        (0, 0, 0), (order, 0, 0), (0, order, 0), (0, 0, order)]]
+
+    if order > 1:
+        for i in range(1, order):
+            cell.append(coord_to_vertex(0, order - i, i))
+        for i in range(1, order):
+            cell.append(coord_to_vertex(order - i, 0, i))
+        for i in range(1, order):
+            cell.append(coord_to_vertex(order - i, i, 0))
+        for i in range(1, order):
+            cell.append(coord_to_vertex(0, 0, i))
+        for i in range(1, order):
+            cell.append(coord_to_vertex(0, i, 0))
+        for i in range(1, order):
+            cell.append(coord_to_vertex(i, 0, 0))
+
+        for j in range(1, order):
+            for i in range(1, order - j):
+                cell.append(coord_to_vertex(order - i - j, i, j))
+        for j in range(1, order):
+            for i in range(1, order - j):
+                cell.append(coord_to_vertex(0, i, j))
+        for j in range(1, order):
+            for i in range(1, order - j):
+                cell.append(coord_to_vertex(i, 0, j))
+        for j in range(1, order):
+            for i in range(1, order - j):
+                cell.append(coord_to_vertex(i, j, 0))
+
+        for k in range(1, order):
+            for j in range(1, order - k):
+                for i in range(1, order - j - k):
+                    cell.append(coord_to_vertex(i, j, k))
+
+    domain = ufl.Mesh(ufl.VectorElement(
+        "Lagrange", ufl.Cell("tetrahedron", geometric_dimension=3), order))
+
+    mesh = create_mesh(MPI.COMM_WORLD, [cell], points, domain)
+    for i in range(mesh.topology.dim):
+        mesh.topology.create_entities(i)
+    md = {"quadrature_degree": 10}
+    measures = (ufl.ds(mesh, metadata=md), ufl.dx(mesh, metadata=md))
+    dimensions = (mesh.topology.dim - 1, mesh.topology.dim)
+    for dim, dC in zip(dimensions, measures):
```

Comment:
# Check that creating a submesh of single cell mesh, consisting of:
# 1. The cell
# 2. The facets of the cell
# Gives the correct computation of: volume (case 1) or surface area (case 2)